extractors/log/tests/integration.rs

```diff
@@ -0,0 +1,143 @@
+    log::info!("exe_path={:?}", corepc_node::exe_path());
+
+    if let Ok(exe_path) = corepc_node::exe_path() {
+        log::info!("Using bitcoind at '{}'", exe_path);
+        return corepc_node::Node::with_conf(exe_path, &conf).unwrap();
+    }
+
+    log::info!("Trying to download a bitcoind..");
+    return corepc_node::Node::from_downloaded_with_conf(&conf).unwrap();
+}
+
+fn setup_two_connected_nodes() -> (corepc_node::Node, corepc_node::Node) {
+    // node1 listens for p2p connections
+    let mut node1_conf = corepc_node::Conf::default();
+    node1_conf.p2p = corepc_node::P2P::Yes;
+    let node1 = setup_node(node1_conf);
+
+    // node2 connects to node1
+    let mut node2_conf = corepc_node::Conf::default();
+    node2_conf.p2p = node1.p2p_connect(true).unwrap();
+    let node2 = setup_node(node2_conf);
+
+    (node1, node2)
+}
+
+async fn check(check_expected: fn(Event) -> ()) {
+    setup();
+    let (node1, _node2) = setup_two_connected_nodes();
+    let nats_server = NatsServerForTesting::new().await;
+    let (shutdown_tx, shutdown_rx) = watch::channel(false);
+
+    let log_extractor_handle = tokio::spawn(async move {
+        let node1_workdir = node1.workdir().to_str().unwrap().to_string();
+        let log_path = format!("{}/regtest/debug.log", node1_workdir);
+        let pipe_path = format!("{}/bitcoind_pipe", node1_workdir);
+        spawn_pipe(log_path, pipe_path.clone());
+
+        let args = make_test_args(nats_server.port, pipe_path.to_string());
+
+        log_extractor::run(args, shutdown_rx.clone())
+            .await
+            .expect("log extractor failed");
+    });
+
+    let nc = async_nats::connect(format!("127.0.0.1:{}", nats_server.port))
+        .await
+        .unwrap();
+    let mut sub = nc.subscribe("*").await.unwrap();
+
+    while let Some(msg) = sub.next().await {
+        let unwrapped = EventMsg::decode(msg.payload).unwrap();
+        if let Some(event) = unwrapped.event {
+            check_expected(event);
+            break;
+        }
+    }
+
+    shutdown_tx.send(true).unwrap();
+    log_extractor_handle.await.unwrap();
+}
+
+#[tokio::test]
+async fn test_integration_logextractor_log_events() {
+    println!("test that we receive log events");
+
+    check(|event| {
+        let mut received_events = false;
+        match event {
+            Event::LogExtractorEvent(r) => {
+                if let Some(ref e) = r.event {
+                    match e {
+                        log_event::Event::UnknownLogMessage(unknown_log_message) => {
+                            received_events = true;
+                            assert!(unknown_log_message.raw_message.len() > 0);
+                        }
+                        _ => (),
+                    }
+                }
+            }
+            _ => panic!("unexpected event {:?}", event),
+        };
+
+        assert!(received_events, "Did not receive any log events");
```

Comment:
this test is just a POC hehe, probably there will be more tests related to string format... using bcli or another node to trigger something...